### PR TITLE
docs(new): Add TSDoc to Coverage class

### PR DESCRIPTION
**Pull request by hashseed**

### CODE REVIEW

File: src/common/Coverage.ts [hunk 1]
```diff
@@ -36,6 +91,17 @@ export class Coverage {
     this._cssCoverage = new CSSCoverage(client);
   }
 
+  /**
+   * @param options - Set of configurable options for coverage.
+   * @param options.resetOnNavigation - Whether to reset coverage on every navigation. @defaultValue true.
```
Comment (hashseed) [inline src/common/Coverage.ts]: This doesn't seem to be the right syntax. How do we document a bag of options?

File: src/common/Coverage.ts [hunk 2]
```diff
@@ -21,12 +21,67 @@ import { CDPSession } from './Connection';
 
 import { EVALUATION_SCRIPT_URL } from './ExecutionContext';
 
+/**
+ * The CoverageEntry class represents one entry of the coverage report.
+- `url` <[string]> StyleSheet URL
+- `text` <[string]> StyleSheet content
+- `ranges` <[Array]<[Object]>> StyleSheet ranges that were used. Ranges are sorted and non-overlapping.
+  - `start` <[number]> A start offset in text, inclusive
+  - `end` <[number]> An end offset in text, exclusive
```
Comment (jackfranklin) [inline src/common/Coverage.ts]: there's no need to document these here - the comments you've added below will do the trick.

What you do need is to mark this as `@public` and also `export` the interface too.

File: src/common/Coverage.ts [hunk 3]
```diff
@@ -21,12 +21,67 @@ import { CDPSession } from './Connection';
...
+   * The covered range as start and end positions.
+   */
   ranges: Array<{ start: number; end: number }>;
 }
 
+/**
+ * The Coverage class provides methods to gathers information about parts of
+ * JavaScript and CSS that were used by the page.
+ * 
+ * @example
```
Comment (jackfranklin) [inline src/common/Coverage.ts:77]: I think the `@remark` should go before the example so the summary (the stuff pre-remark) is nice and concise.

File: src/common/Coverage.ts (hunk 1 above)
Comment (jackfranklin) [reply src/common/Coverage.ts]: The best fix is to extract an interface for these options so you can document them properly. Else TSDoc doesn't handle it very well :(

Comment (hashseed) [reply src/common/Coverage.ts]: Done.

File: src/common/Coverage.ts (hunk 2 above)
Comment (hashseed) [reply src/common/Coverage.ts]: Done.

File: src/common/Coverage.ts (hunk 3 above)
Comment (hashseed) [reply src/common/Coverage.ts:77]: Done.

**Comment by hashseed** [comment]
Please re-review :)

**Comment by jackfranklin** [comment]
@hashseed LGTM! I think after you merged in `main` you need to re-generate the docs and then CI will be happy.

**Review by jackfranklin — APPROVED** [review]

**Pull request merged**